client/components/LearnerViewDirective.js

```diff
@@ -553,9 +588,28 @@ tie.directive('learnerView', [function() {
             } else if (syntaxErrorIndex === null) {
               $scope.syntaxErrorString = '';
               $scope.syntaxErrorFound = false;
+
+              // Updating reinforcement bullets only if no syntax errors
+              reinforcement.clear();
+              for (var caseList in reinforcementObject.passedList) {
+                if (reinforcementObject.passedList[caseList]) {
```

Comment:
I was using a style similar to congratulatoryFeedback defined on line 515. The reinforcement object we get from the service is always a new one, from which entries are taken and the main stored object is updated.